pkg/storegateway/indexheader/reader_pool.go

```diff
@@ -152,16 +152,15 @@ func (p *ReaderPool) onLazyReaderClosed(r *LazyBinaryReader) {
 	delete(p.lazyReaders, r)
 }
 
-// LoadedBlocks returns a new map of lazy-loaded block IDs and the last time they were used in milliseconds.
-func (p *ReaderPool) LoadedBlocks() map[ulid.ULID]int64 {
+func (p *ReaderPool) LoadedBlocks() map[ulid.ULID]struct{} {
```

Comment:
Should we just return a slice here? The only place that calls this is `Snapshotter.PersistLoadedBlocks`, and it doesn't need a map.